### PR TITLE
Docs role handbook clean up

### PR DESCRIPTION
**Pull request by annajung**
#### What type of PR is this:
/kind cleanup

#### What this PR does / why we need it:
- Add a table of content on the top of the docs handbook to allow quicker access to content
- Fix broken links
- Remove unused google groups
- Add addition instructions on how to become a member of google groups managed in github

#### Which issue(s) this PR fixes:
None

#### Special notes for your reviewer:
With 1.20 release starting, I'm hoping I can continuously improve this handbook!

/assign @jimangel @savitharaghunathan

### CODE REVIEW

File: release-team/role-handbooks/docs/README.md [hunk 1]
```diff
@@ -78,19 +138,20 @@ For each release, the schedule with deliverables is added to the release directo
...
 
-- ⚠️ Join these Google groups:
+- ⚠️  Join these Google groups:
     - [kubernetes-sig-release](https://groups.google.com/forum/#!forum/kubernetes-sig-release)
-    - [kubernetes-sig-leads](https://groups.google.com/forum/#!forum/kubernetes-sig-leads)
     - [kubernetes-sig-docs](https://groups.google.com/forum/#!forum/kubernetes-sig-docs)
-    - [kubernetes-milestone-burndown](https://groups.google.com/forum/#!forum/kubernetes-milestone-burndown)
     - [kubernetes-dev](https://groups.google.com/forum/#!forum/kubernetes-dev)
-    - [kubernetes-release-team](https://groups.google.com/a/kubernetes.io/g/release-team) # for enhancement google sheets access!
+    - [kubernetes-release-team](https://groups.google.com/forum/#!forum/kubernetes-release-team)
```
Comment (hasheddan) [inline release-team/role-handbooks/docs/README.md:151]:
```suggestion
    - [kubernetes-release-team](https://groups.google.com/a/kubernetes.io/g/release-team)
```

File: release-team/role-handbooks/docs/README.md [hunk 2]
```diff
@@ -78,19 +138,20 @@ For each release, the schedule with deliverables is added to the release directo
...
+        - ⚠️  List of members for this group is managed in git. Please create a PR against `kuberenetes/sig-release` repo to include your email under `release-team` group in [`sig-release/group.yaml`](https://github.com/kubernetes/k8s.io/blob/master/groups/sig-release/groups.yaml) 
+    - [kubernetes-release-team-shadows](https://groups.google.com/forum/#!forum/kubernetes-release-team-shadow)
```
Comment (hasheddan) [inline release-team/role-handbooks/docs/README.md:153]:
```suggestion
    - [kubernetes-release-team-shadows](https://groups.google.com/a/kubernetes.io/g/release-team-shadows)
```

**Review by hasheddan — COMMENTED** [review]
A few of the google groups links just recently changed :)

xref https://github.com/kubernetes/k8s.io/pull/1250

File: release-team/role-handbooks/docs/README.md (hunk 1 above)
Comment (annajung) [reply release-team/role-handbooks/docs/README.md]: Fixed! Thank you :)

File: release-team/role-handbooks/docs/README.md (hunk 2 above)
Comment (annajung) [reply release-team/role-handbooks/docs/README.md]: Fixed! Thank you :)

**Review by savitharaghunathan — COMMENTED** [review]
Hi, @annajung 👋 Thanks for cleaning up the docs role handbook. 

/lgtm

Please squash the commits into one when you get a moment, I will add `lgtm` again once all the commits are squashed.

**Comment by annajung** [comment]
been squashed! Thanks @savitharaghunathan

**Review by savitharaghunathan — COMMENTED** [review]
/lgtm

**Comment by jimangel** [comment]
/lgtm
/approve
/hold cancel

Thanks!

**Pull request merged**